pxtlib/analytics.ts

```diff
@@ -33,6 +33,13 @@ namespace pxt.analytics {
 
         const te = pxt.tickEvent;
         pxt.tickEvent = function (id: string, data?: Map<string | number>, opts?: TelemetryEventOptions): void {
+            // Log tick to console if "consoleticks" url param is present.
+            // We do this simple check first (rather than the full parse) to minimize overhead when it's not present, which is most of the time.
+            if (window && /consoleticks=/.test(window.location.href))
```

Comment:
I'd prefer this were controlled by a flag set by the application, e.g. `pxt.analytics.consoleTicks = true`. How the flag is set is up to the app.

Reply:
Here's where we have a few similar flags set to this:

https://github.com/microsoft/pxt/blob/stable8.5/webapp/src/app.tsx#L5539

Reply:
@eanders-ms - just wanted to clarify, do you mean it should still be enabled/disabled via a query parameter, but that query parameter should then set a flag (like `pxt.analytics.consoleTicks`), or do you mean we should just have a flag hardcoded to false which devs can set to true if they want to debug? Or something else entirely?

Reply:
pxtlib should be agnostic to how the flag is set. Leave that to the app. The webapp would parse the query param and set the flag. The cli would read a command line arg.